src/popup/pages/Names/Claim.vue

```diff
@@ -72,94 +73,99 @@ import CheckBox from '../../components/CheckBox.vue';
 
-export default {
+export default defineComponent({
   name: 'Claim',
   components: {
     InputField,
     CheckBox,
     BtnMain,
     BtnHelp,
   },
-  data: () => ({
-    AENS_DOMAIN,
-    name: '',
-    loading: false,
-    autoExtend: false,
-    maxNameLength: AENS_NAME_MAX_LENGTH - AENS_DOMAIN.length,
-  }),
-  computed: {
-    isNameValid() {
-      return this.name && checkAensName(`${this.name}${AENS_DOMAIN}`);
-    },
-    nameFee() {
-      return convertToken(
-        TxBuilderHelper.getMinimumNameFee(`${this.name}${AENS_DOMAIN}`),
-        -MAGNITUDE,
-      ).toFixed(4);
-    },
-  },
-  methods: {
-    async claim() {
-      if (!await this.$validator.validateAll()) return;
+  setup(props, { root }) {
+    const name = ref('');
+    const autoExtend = ref(false);
+    const loading = ref(false);
+    const maxNameLength = AENS_NAME_MAX_LENGTH - AENS_DOMAIN.length;
+
+    const isNameValid = computed(() => name.value && checkAensName(`${name.value}${AENS_DOMAIN}`));
+
+    const nameFee = computed(() => convertToken(
+      TxBuilderHelper.getMinimumNameFee(`${name.value}${AENS_DOMAIN}`),
+      -MAGNITUDE,
+    ).toFixed(4));
+
+    const { getSdk, isSdkReady } = useSdk({ store: root.$store });
+
+    async function claim() {
+      if (!await (root as any).$validator.validateAll()) return;
 
-      const { getSdk } = useSdk({ store: this.$store });
       const { openDefaultModal } = useModals();
-      const { activeAccount } = useAccounts({ store: this.$store });
+      const { activeAccount } = useAccounts({ store: root.$store });
 
-      const name = `${this.name}${AENS_DOMAIN}`;
-      const nameEntry = await this.sdk.api.getNameEntryByName(name).catch(() => false);
+      const sdk = await getSdk();
+
+      const fullName = `${name.value}${AENS_DOMAIN}`;
+      const nameEntry = await sdk.api.getNameEntryByName(fullName).catch(() => false);
 
       if (nameEntry) {
         openDefaultModal({
-          title: this.$t('modals.name-exist.msg'),
+          title: root.$t('modals.name-exist.msg'),
         });
       } else {
-        this.loading = true;
+        loading.value = true;
         let claimTxHash;
-        const sdk = await getSdk();
 
         try {
-          const { salt } = await sdk.aensPreclaim(name);
-          claimTxHash = (await sdk.aensClaim(name, salt, { waitMined: false })).hash;
-          if (this.autoExtend) {
-            this.$store.commit('names/setPendingAutoExtendName', name);
+          const { salt } = await sdk.aensPreclaim(fullName);
+          claimTxHash = (await sdk.aensClaim(fullName, salt, { waitMined: false })).hash;
+          if (autoExtend.value) {
+            root.$store.commit('names/setPendingAutoExtendName', fullName);
           }
-          this.$router.push({ name: ROUTE_ACCOUNT_DETAILS_NAMES });
-        } catch (e) {
+          root.$router.push({ name: ROUTE_ACCOUNT_DETAILS_NAMES });
+        } catch (e: any) {
```

Comment:
```suggestion
        } catch (e: Exception) {
```

Reply:
Every other place in the project has `: any` in such case, i don't want to solve that in this particular issue.
```
Catch clause variable type annotation must be 'any' or 'unknown' if specified.Vetur(1196)
interface Error
```

Reply:
Yeah, I see that. Perhaps we should start writing custom exceptions. This is not the right moment as too many refactorings already on it's way.